source/local/local_test.go

```diff
@@ -33,3 +36,53 @@ func TestLocalSource(t *testing.T) {
 	assert.Empty(t, l)
 
 }
+
+func TestGetExpirationDate(t *testing.T) {
+	expectedFeaturesLen := 2
+	pwd, _ := os.Getwd()
+	featureFilesDir = filepath.Join(pwd, "testdata/features.d")
+
+	features, err := getFeaturesFromFiles()
+
+	assert.NoError(t, err)
+	assert.Equal(t, expectedFeaturesLen, len(features))
+}
+
+func TestParseDirectives(t *testing.T) {
+	testCases := []struct {
+		name        string
+		directive   string
+		expectedErr error
+	}{
+		{
+			name:        "valid directive",
+			directive:   "# +expiry-time=2080-07-28T11:22:33Z",
+			expectedErr: nil,
+		},
+		{
+			name:        "unknown directive",
+			directive:   "# +random-key=random-value",
+			expectedErr: ErrorUnknownDirective,
+		},
+		{
+			name:        "unknown directive",
+			directive:   "# +random-key=random-value",
+			expectedErr: ErrorUnknownDirective,
+		},
+		{
+			name:        "unkown directive structure",
```

Comment:
```suggestion
			name:        "unknown directive format",
```
(Or maybe "invalid" would suit better here than "unknown".)